test/unit/ngsiv2/MQTT_receive_measures-test.js

```diff
@@ -0,0 +1,259 @@
+            .reply(200, '{}');
+
+        iotaJson.start(config, function() {
+            request(provisionOptions, function(error, response, body) {
+                done();
+            });
+        });
+    });
+
+    afterEach(function(done) {
+        nock.cleanAll();
+        mqttClient.end();
+
+        async.series([
+            iotAgentLib.clearAll,
+            iotaJson.stop
+        ], done);
+    });
+
+    describe('When a new multiple measure arrives to the MQTT Topic', function() {
+        beforeEach(function() {
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/Second%20MQTT%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/multipleMeasuresJsonTypes.json'))
+                .reply(204);
+        });
+        it('should send its value to the Context Broker', function(done) {
+            var values = {
+                humidity: '32',
+                temperature: '87',
+                luminosity: 10,
+                pollution: 43.4,
+                configuration: {firmware: {verson: '1.1.0', hash: 'cf23df2207d99a74fbe169e3eba035e633b65d94'}},
+                tags: ['iot', 'device'],
+                enabled: true,
+                alive: null
+            };
+
+            mqttClient.publish('/1234/MQTT_2/attrs', JSON.stringify(values), null, function(error) {
+                setTimeout(function() {
+                    contextBrokerMock.done();
+                    done();
+                }, 100);
+            });
+        });
+    });
+
+    describe('When a new multiple measure arrives for an unprovisioned device', function() {
+        beforeEach(function(done) {
+
+            // Note /v1/updateContext response is not processed by IOTA so its content is irrelevant,
+            // as far as it is a 200 OK
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext')
+                .reply(200, '{}');
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities/TheLightType:MQTT_UNPROVISIONED/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/unprovisionedDevice.json'))
+                .reply(204);
+
+            request(groupCreation, function(error, response, body) {
+                done();
+            });
+        });
+        it('should send its value to the Context Broker', function(done) {
+            var values = {
+                humidity: '32',
+                temperature: '87'
+            };
+
+            mqttClient.publish('/KL223HHV8732SFL1/MQTT_UNPROVISIONED/attrs', JSON.stringify(values), null,
```

Comment:
JSON_UNPROVISIONED?

Reply:
Fixed in 10969e0f085eab0d57d2bd42df675f732c2f7629